object/src/mem.rs

```diff
@@ -916,6 +927,15 @@ where
         unreachable!()
     }
 
+    pub fn convert_to_utf8(&mut self) {
```

Comment:
Ah yes, I'll add documentation.

> Maybe we could have a more generic convert_charset alongside this one (SpecificCharacterSet)?

I was thinking about that, but I wasn't sure if it made sense. Going to UTF-8 is special in a sense, because it makes the internal representation the same as dicom charset. In that way it makes reasoning about the charset easy.

I wasn't sure about the use-case for going the other way. It might also be less trivial, because we are not sure that the conversion can be done (we'd have to check for instance for any non `ISO_IR 100` characters). The check could of course be postponed until we write the dataset, which has some downsides on its own.

TLDR; the implementation to change the charset itself would be easy, but there might be consequences that are harder to work with.